### DISCUSSION TITLE
为什么我的请求失败

### DISCUSSION THREAD

**Discussion by yuyi2439** [howmanybots/onebot#105] [问答]
我向onebot发送了一串json：
{
    "action": "send_private_msg",
    "params": {
        "user_id": 2961354039,
        "message": "你好"
    },
    "echo": "123"
}
但是onebot回复是：
 “HTTP/1.1 400 Bad Request
Connection: close

”
1400：  如果 POST 请求的正文格式不正确，状态码为 400

**Comment by stdrc** [comment]
检查一下有没有 `Content-Type: application/json` header？

**Reply by yuyi2439** [reply]
刚刚加上了，但是返回的还一样

**Reply by yuyi2439** [reply]
应该是少了请求头

**Reply by zhzhongshi** [reply]
这是用ws发送的？

**Comment by xiguasamall** [comment]
看起来像ws的body
但是为何回复的是 HTTP/1.1

**Reply by yuyi2439** [reply]
是ws的body
但是我没有看到标准里的ws头

**Reply by yuyi2439** [reply]
请求头应该是什么
